### PR TITLE
Status error presentation with details

### PR DESCRIPTION
**Pull request by werkt**
Remote Execution Status messages embedded in ExecuteResponses are
extremely capable vehicles for conveying the nature of an error, and
informing a user of further steps to take to remediate it. This change
expands the presentation of these response Statuses, and brings all of
the error details to light, by default instead of requiring
--verbose_failures to investigate any details of a remote execution
problem.

The interpretation of precondition failures to highlight retriable
responses has been expanded to ignore benign details that might be
included in a response.

SpawnResult error message composition has been simplified substantially,
without any special behavior for 'Remote' errors, and a removal of a
duplicate message printout incurred in the wake of succcessive @janakr
and @olaola changes. Failure messages are now implied to be present in
all spawn result failure reporting exactly once, and the failureMessage
of a SpawnResult is implied to be the parameter to getDetailMessage.

An example error presentation is as follows (including the modifications
to SpawnResult's output formatting):

```
ERROR: /home/werkt/dev/test/BUILD:22:10: Linking test failed: (Exit 34): Remote Execution Failure:
Failed Precondition: Action 4223ab2cc114385110714243a0b4a88cc743f2169b5be7d4d438a6bbba4f529f/142 is invalid
  Resource Info: type.googleapis.com/google.longrunning.Operation: name='shard/operations/9335fef2-184b-4d26-9a6f-2f27cebe7527', owner='tool_invocation_id:4b4bf7b1-fadd-44fd-99be-a234e7c26fc4,correlated_invocation_id:dc88325a-9317-48c0-9013-b3bb8b7a458f'
  Precondition Failure:
    (MISSING) bazel-out/k8-fastbuild/bin/test: 7872: An output could not be uploaded because it exceeded the maximum size of an entry
Target //:test failed to build
```

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/remote/ExecuteRetrier.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    com.google.rpc.Status status = StatusProto.fromThrowable(e);
+    if (status == null || status.getDetailsCount() == 0) {
+      return false;
+    }
+    boolean fullyRetriable = false;
+    boolean failedPrecondition = status.getCode() == Code.FAILED_PRECONDITION.value();
+    for (Any detail : status.getDetailsList()) {
+      if (detail.is(RetryInfo.class)) {
+        // server says we can retry, regardless of other details
+        fullyRetriable = true;
```
Comment (coeuvre) [inline src/main/java/com/google/devtools/build/lib/remote/ExecuteRetrier.java:97]: `break` here?

**Review by coeuvre — COMMENTED** [review]
Thanks!

File: src/main/java/com/google/devtools/build/lib/remote/ExecuteRetrier.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    RetryInfo getRetryInfo(Exception e) {
+      RetryInfo retryInfo = RetryInfo.getDefaultInstance();
+      com.google.rpc.Status status = StatusProto.fromThrowable(e);
+      if (status != null) {
+        for (Any detail : status.getDetailsList()) {
+          if (detail.is(RetryInfo.class)) {
+            try {
+              retryInfo = detail.unpack(RetryInfo.class);
```
[the remark this answers is not in the document]
Comment (werkt) [reply src/main/java/com/google/devtools/build/lib/remote/ExecuteRetrier.java:61]: I want the retryInfo to have a deterministic last-specified behavior. Not that I want to see multiple from the service, but if it does, the last one in the list should be effective.

File: src/main/java/com/google/devtools/build/lib/remote/ExecuteRetrier.java (hunk 1 above)
Comment (werkt) [reply src/main/java/com/google/devtools/build/lib/remote/ExecuteRetrier.java:97]: Early return here - nothing after it is effective, and coming up with whether the precondition failure is effective is meaningless. Thanks!

File: src/main/java/com/google/devtools/build/lib/remote/ExecuteRetrier.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import com.google.rpc.LocalizedMessage;
+import com.google.rpc.RetryInfo;
+import com.google.rpc.RequestInfo;
+import com.google.rpc.ResourceInfo;
+import com.google.rpc.PreconditionFailure;
+import com.google.rpc.PreconditionFailure.Violation;
+import io.grpc.Status.Code;
+import io.grpc.protobuf.StatusProto;
+
+class ExecuteRetrier extends RemoteRetrier {
```
[the remark this answers is not in the document]
Comment (werkt) [reply src/main/java/com/google/devtools/build/lib/remote/ExecuteRetrier.java:33]: Hope a single line is sufficient here, didn't see much else on the other classes I looked at.

**Review by coeuvre — APPROVED** [review]
